core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -110,7 +115,10 @@ public <T extends RESTResponse> T execute(RESTCatalogAdapter.HTTPMethod method,
 
     this.restCatalog = new RESTCatalog(context, (config) -> adaptor);
     restCatalog.setConf(conf);
-    restCatalog.initialize("prod", ImmutableMap.of(CatalogProperties.URI, "ignored", "credential", "catalog:12345"));
+    restCatalog.initialize("prod", ImmutableMap.of(
+        CatalogProperties.URI, "ignored", "credential", "catalog:12345",
+        String.format("header.%s", HTTPClientFactory.CLIENT_GIT_COMMIT_SHORT_HEADER), IcebergBuild.gitCommitShortId(),
+        String.format("header.%s", HTTPClientFactory.CLIENT_VERSION_HEADER), IcebergBuild.version()));
```

Comment:
These don't really need to be set because `HTTPClientFactory` isn't used here - the underlying JDBC client is what builds this.

We can however still override the version etc by sending back a configuration property of `header.X-Client-Version` with the desired value to be used as the value (though I don't see any reason to do that).